src/store/modules/swap/ethBancor.ts

```diff
@@ -1851,43 +1850,33 @@ export class EthBancorModule
           })
         ).catch(e => {
           console.warn("Error fetching ROIs", e);
-        }),
-        Promise.all(
-          allPositions.map(async position => {
-            const currentPoolBalances = await this.fetchRelayBalances({
-              poolId: position.poolToken
-            });
+        })
+      ]);
 
-            const [
-              depositedReserve,
-              opposingReserve
-            ] = sortAlongSide(
-              currentPoolBalances.reserves,
-              reserve => reserve.contract,
-              [position.reserveToken]
-            );
-            const rate0 = new BigNumber(position.reserveRateN)
-              .div(position.reserveRateD)
-              .toString();
-
-            const feeAmountWei = calculatePositionFees(
-              position.poolAmount,
-              currentPoolBalances.smartTokenSupplyWei,
-              position.reserveAmount,
-              depositedReserve.weiAmount,
-              opposingReserve.weiAmount,
-              rate0
-            );
+      const poolReserveIds = allPositions.map(position => {
+        return { poolId: position.poolToken, reserveId: position.reserveToken };
+      });
 
-            const shrunk = shrinkToken(feeAmountWei, 18);
+      const uniquePoolReserveIds = uniqWith(poolReserveIds, isEqual);
```

Comment:
This is totally legit, just be mindful of any potential casing differences in other places. I think it's safe to assume the matching cases from this request. Sometimes I'll do a bloody ```(a,b) => compareString(a.poolId, b.poolId) && compareString(a.reserveId, b.reserveId)``` to be absolutely sure.